### ISSUE TITLE
As Pho, I want to get notified if any data of a group is missing

### ISSUE THREAD

**Issue by evaklopp** [buzzn/scrum#22]
**PR**
- https://github.com/buzzn/admin-app/pull/248
- https://github.com/buzzn/core/pull/1302
---
**Story**
As PhO I want to get notified if any data of a group is missing on the screen, where the missing data occurs. 

---
- excluded story from #5 
- connected to #52 where the design happens

---
**XD**
no for the moment

---
**DoD**
- [x] show business objects with missing or inconsistent data ~~in the right todo-bar~~

exclude:
- highlight fields and list with incomplete data where it is applicable

---
**technical description**

**Comment by mkristian** [comment]
@evaklopp since I added the bank_account to a field to be there for localpool to be called complete. I wonder if all the addresses:
* localpool.address
* localpool.owner.address (both cases: person and organization)
* localpool.owner.contact.address
is also a mandatory entity for a localpool

cc/ @dottorer

**Comment by dottorer** [comment]
@mkristian I would think  the following is a good set of rules:
1. a local group must have an address
2. an owner (both cases) must have an address (i.e. mailing address for letters, etc.)
3. a contact person might NOT have an address since it could be sufficient to contact by email or phone.

my 2 cts.

**Comment by mkristian** [comment]
@scsirdx once my PR is merged this is done from my side and your side ?

**Comment by mkristian** [comment]
@scsirdx I see you moved it already but staging is not deployed to the last PR is not visible. but it just adds more to list as not all entities have addreses

**Comment by phillipoertel** [comment]
@mkristian @scsirdx I'm not sure I get your comment. 
To recap, please make sure everything you developed in a ticket (see the definition of done) is visible/testable on staging, before you move a ticket to "ready to test". I think you could support her by adding some links to localpools which show the incompleteness validations, so she doesn't have to search for them herself. Ideally you would also list which incompleteness checks we currently do, as that isn't described anywhere yet, IMHO.

**Comment by scsirdx** [comment]
@mkristian @phillipoertel  It seems that i misunderstood a ticket.
We discussed previously that incompleteness is now "logical" and not field based. So i fixed all frontend issues and pushed it to "test".
Incompleteness is now displayed for the group only, because all other screens are changed dramatically, and we don't have a proper concept for the notifications.

Taking into account this notes user will see only notifications on the right block and in groups list. So for testing it will no matter too much how many incompleteness issues will be displayed. And we don't have any errors translations right now. So its more like POC ticket. And design + translations will be handled in next story.
Am i correct?

**Comment by phillipoertel** [comment]
@scsirdx you're right, it's a proof of concept ticket which we need to refine, and we can fine-tune there. 
Still, adding information about which checks were built in this ticket will make our Product Owner's life easier, and our process smoother. For one, she needs to know what to test here (and on which pools she can check it), and she also needs to know which checks (not) to specify in future tickets, because we already have them. Since @mkristian and you just built these checks anyways, it's very easy for you to add this helpful info here, IMHO.

**Comment by scsirdx** [comment]
I agree.
So we should add comment that `highlight fields and list with incomplete data where it is applicable` is irrelevant and provide incompleteness list?
Right now i'll move this ticket back.

**Comment by mkristian** [comment]
@scsirdx I will add the checks implemented and also deploy to staging - just not tonight but tomorrow morning

@phillipoertel the links I can provide but they are only valid until we push the DB import next time.

**Comment by mkristian** [comment]
@evaklopp it is on staging. just go into any group and you will see at least

https://staging-admin.buzzn.io/localpools/26e67d8e-057f-4c02-9933-949c24f5ba19/settings
https://staging-admin.buzzn.io/localpools/30afe242-ffd9-4dc3-82ba-a63b35a5a13b/settings
https://staging-admin.buzzn.io/localpools/8a3974a0-31a7-4e0b-92f8-3035786d57e3/settings

there is more checks on addresses but they seems to be all complete.

note that the links are valid until our next deploy

**Comment by mkristian** [comment]
@evaklopp new links after the staging deploy:

https://staging-admin.buzzn.io/localpools/fb244995-c78e-437f-a94d-2a8cd550e6a9/settings?devmode=true
https://staging-admin.buzzn.io/localpools/b6333b63-e682-4f2e-94a7-69c477e25efa/settings
https://staging-admin.buzzn.io/localpools/97c0eccf-150d-48b1-96d3-8754041527d2/settings

if you want to turn off the devmode again:
https://staging-admin.buzzn.io/localpools/fb244995-c78e-437f-a94d-2a8cd550e6a9/settings?devmode=false

**Comment by phillipoertel** [comment]
@evaklopp the technical part is ready but hidden behind a "devmode" setting. Please bring a follow-up ticket to finalize the UX into one of the next sprint

**Comment by evaklopp** [comment]
@phillipoertel I already opened #52 to adress the design topic. 

@scsirdx short question: the errors/missing data I only see in the list view, not in the tiles view, right? 

@mkristian If I open your links, I get 404 errors. But If I simply add the /settings?devmode=true I can see the errors. Is that a correct workaround?

**Comment by phillipoertel** [comment]
@evaklopp @mkristian with every new beekeeper import the localpool links/URLs change on the staging system. We currently do that at least once a day. 
So for now it would be better to just name the localpools we want to test with, as the URLs are unreliable. @mkristian opened #56 with a suggestion to address the problem.

**Comment by evaklopp** [comment]
hooray
